comm/p2p/libp2p.go

```diff
@@ -54,15 +55,18 @@ func (c Libp2pCommunication) CloseSession(sessionID string) {
 }
 
 func (c Libp2pCommunication) Broadcast(
+	ctx context.Context,
 	peers peer.IDSlice,
 	msg []byte,
 	msgType comm.MessageType,
 	sessionID string,
 ) error {
+	span := traceapi.SpanFromContext(ctx)
 	hostID := c.h.ID()
 	wMsg := comm.WrappedMessage{
 		MessageType: msgType,
 		SessionID:   sessionID,
+		TraceID:     span.SpanContext().TraceID().String(),
```

Comment:
Why do we need this?

Reply:
It seems better to set custom traceID than to send trace id in the message.

Reply:
Possibly if we set it to msg.ID() it would also unify the listen and route spans.

Reply:
we do not have msg.ID() (we are operating of proposalsHash on this lvl of abstraction) in here, this traceID is used to share the trace among relayers in TSS process.
I also do not want really to use sessionID bcs then all the Relayers will be joined under one trace, which I want to avoid, since as I showed in demo, for example, relayers that are not participating in particular TSS process (bcs we do not need all relayers to sign) do not need to get into the trace. 
Although this might be investigated in details in future to observe some patterns and maby eventually add everething under one trace, but not now probably

Reply:
But right now you are sharing traceID anyway.
Even if you don't want to share it, then just adding relayerID to it would make it that way.
MsgID could be easily passed with context.
For sure we don't need traceID in the message directly.

Reply:
yeah I am sharing traceID with sedning it over the libp2p, is the same ID will be added as traceID it will automatically consider them all the same trace

Reply:
I specifically don't want traceId in the message

Reply:
Sending traceID's in requests among services is a common practice. Message is our package that is getting shared among services. 
Theoretically, there might be a better way to add TraceID to the comm headers or other MetaData, I would agree to change this in future updates by creating an issue bcs generally I agree with you.